docs/en/02_Developer_Guides/14_Files/03_File_Security.md

```diff
@@ -0,0 +1,369 @@
+
+
+	:::yaml
+	File: 
+	  allowed_extensions: 
+	    - 7zip 
+	    - xzip
+
+
+<div class="warning" markdown="1">
+Any file not included in this config, or in the default list of extensions, will be blocked from
+any requests to the assets directory. Invalid files will be blocked regardless of whether they
+exist or not, and will not invoke any PHP processes.
+</div>
+
+### Configuring: Secure file headers
+
+In certain situations, it's necessary to customise HTTP headers required either by
+intermediary caching services, or by the client, or upstream caches.
+
+When a secure file is served it will also be transmitted with all headers defined by the
+`SilverStripe\Filesystem\Flysystem\FlysystemAssetStore.file_response_headers` config.
+You can customise this with the below config:
+
+
+    :::yaml
+    SilverStripe\Filesystem\Flysystem\FlysystemAssetStore:
+      file_response_headers:
+        Pragma: 'no-cache'
+
+
+### Configuring: Archive behaviour
+
+By default, the default extension `AssetControlExtension` will control the disposal of assets
+attached to objects when those objects are deleted. For example, unpublished versioned objects
+will automatically have their attached assets moved to the protected store. The deletion of 
+draft or unversioned objects will have those assets permanantly deleted (along with all variants).
+
+In some cases, it may be preferable to have any deleted assets archived for versioned dataobjects,
+rather than deleted. This use more disk storage, but will allow the full recovery of archived
+records and files.
+
+This can be applied to DataObjects on a case by case basis by setting the `archive_assets`
+config to true on that class. Note that this feature only works with dataobjects with
+the `Versioned` extension.
+
+
+    :::php
+    class MyVersiondObject extends DataObject {
+        /** Enable archiving */
+        private static $archive_assets = true;
+        /** Versioned */
+        private static $extensions = array('Versioned');
+    }
+
+
+The extension can also be globally disabled by removing it at the root level:
+
+
+    :::yaml
+    DataObject:
+      AssetControl: null
+
+
+### Configuring: Web server settings
+
+If the default server configuration is not appropriate for your specific environment, then you can
+further customise the .htaccess or web.config by editing one or more of the below:
+
+* `Assets_HTAccess.ss`: Template for public permissions on the Apache server.
+* `Assets_WebConfig.ss`: Template for public permissions on the IIS server.
+* `Protected_HTAccess.ss`: Template for the protected store on the Apache server (should deny all requests).
+* `Protected_WebConfig.ss`: Template for the protected store on the IIS server (should deny all requests).
+
+Each of these files will be regenerated on ?flush, so it is important to ensure that these files are
+overridden at the template level, not via manually generated configuration files.
+
+#### Configuring Web Server: Apache server
```

Comment:
Note which `AllowOverride` settings are required. Some setups might've removed `.htaccess` on root level and moved it to `httpd.conf` for performance reasons.